README.md

```diff
@@ -15,72 +15,75 @@ riff System contains three API groups with CustomResourceDefinitions that are th
   - `Stream` - streams of messages
-  - `Processor` - processors apply functions to messages on streams
+  - `Processor` - processors apply functions, containers or images to messages on streams
+  - `KafkaProvider` - kafka based stream provider
 - `knative.projectriff.io/v1alpha1`
   - `Adapter` - adapters map applications, functions or container images into an existing Knative Service or Configuration.
   - `Deployer` - deployers map HTTP requests to applications, functions, containers or images with Knative
 
-### Runtime
+### Runtimes
 
-A `controller` and `webhook` Deployments exist in the `riff-system` namespace to validate and reconcile the riff CRDs.
+Each runtime has an independent manager deployment that reconciles and hosts validation and defaulting webhooks for the CRDs needed by that runtime.
+
+The build component is required by each runtime. 
 
 ### RBAC
 
 Two ClusterRoles are defined to grant access to the riff CRDs.
 
-- `projectriff` - read/write access to all riff CRDs
-- `projectriff-readonly` - read access to all riff CRDs
+- `riff-${component}-role` - read/write access to component's CRDs
+- `riff-${component}-readonly-role` - read access to component's CRDs
 
 These roles are aggregated to the `edit` and `view` ClusterRoles respectively.
 
 See the Kuberneties [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) for more information.
 
 ## Install
 
-riff System is not installed directly for typical use. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
+riff System is typical not installed directly by end users. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
 
 ## Development
 
-After making any changes to source files in `./pkgs/apis` it is necessary to regenerate the API client by running:
+riff System builds on [KubeBuilder](https://www.kubebuilder.io) which makes extensive use of generated code and configuration. After making any changes to source files in `./pkg/apis` or `./pkg/controllers` it is necessary to regenerate the API client and configuration by running:
 
 ```sh
-./hack/update-codegen.sh
+make prepare
 ```
 
-Dependencies are managed with dep. After importing a new package or modifying Gopkg.toml, run:
+To run the unit tests locally:
 
 ```sh
-./hack/update-deps.sh
+make test
 ```
 
-To run the unit tests locally:
+To deploy to a development cluster with [ko](https://github.com/google/ko):
 
 ```sh
-go test ./...
+ko apply -f config/
 ```
 
-To deploy to a development cluster with [ko](https://github.com/google/ko):
+or install a single component:
 
 ```sh
-ko apply -f config/
+ko apply -f config/riff-${component}.yaml
 ```
 
 Additional dependencies must be installed independently into the cluster including:
 
-- riff build templates
-- Knative Build
-- Knative Serving
-- Istio
+- [Istio](https://istio.io) (knative runtime)
```

Comment:
Maybe add "needed by" in the parentheses where appropriate, as this may not be immediately clear.